dotnet/src/Agents/Abstractions/ChatHistoryKernelAgent.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Agents;
+
+/// <summary>
+/// A <see cref="KernelAgent"/> specialization bound to a <see cref="ChatHistoryChannel"/>.
+/// </summary>
+public abstract class ChatHistoryKernelAgent : KernelAgent, IChatHistoryHandler
+{
+    /// <inheritdoc/>
+    protected internal sealed override IEnumerable<string> GetChannelKeys()
+    {
+        yield return typeof(ChatHistoryChannel).FullName;
```

Comment:
This is a good question and something I've spent a lot of time exploring.

By definition, this might imply two different `AgentChat` instances that present in an `AggregatorAgent` that participate in a parent `AgentChat`.  I'm exploring support for these concepts in the `Agent as a Plug-in` task (which also includes `AggregatorAgent`.